pkgs/by-name/ha/handheld-daemon/package.nix

```
@@ -1,63 +1,96 @@
 {
+  lib,
+  python3Packages,
   fetchFromGitHub,
+
+  # dependencies
+  systemd,
   hidapi,
-  kmod,
-  lib,
-  python3,
   toybox,
-  lsof
+  kmod,
+  dbus,
+  lsof,
 }:
-python3.pkgs.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "handheld-daemon";
-  version = "3.6.1";
+  version = "3.6.2";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "hhd-dev";
     repo = "hhd";
     rev = "refs/tags/v${version}";
-    hash = "sha256-IdpSRb66G+WzTv/BC29r2OjO1b4VdWbV6OSzOoiFAO0=";
+    hash = "sha256-W1Ap6yTryBDozKe3aO413Fu0RBul9kEA9ACUTdYyOKM=";
   };
 
-  propagatedBuildInputs = with python3.pkgs; [
+  postPatch =
+    let
+      udevadm = lib.getExe' systemd "udevadm";
+    in
+    ''
+      # handheld-daemon contains a fork of the python module `hid`, so this hook
+      # is borrowed from the `hid` derivation.
+      hidapi=${lib.getLib hidapi}/lib/
+      test -d $hidapi || { echo "ERROR: $hidapi doesn't exist, please update/fix this build expression."; exit 1; }
+      substituteInPlace src/hhd/controller/lib/hid.py \
+        --replace-fail "libhidapi" "$hidapi/libhidapi"
+
+      # The generated udev rules point to /bin/chmod, which does not exist in NixOS
+      substituteInPlace src/hhd/controller/lib/hide.py \
+        --replace-fail "/bin/chmod" "${lib.getExe' toybox "chmod"}" \
+        --replace-fail "udevadm" "${udevadm}"
+
+      substituteInPlace src/hhd/controller/physical/evdev.py \
+        --replace-fail "udevadm" "${udevadm}"
+
+      substituteInPlace src/hhd/controller/physical/imu.py \
+        --replace-fail '"modprobe' '"${lib.getExe' kmod "modprobe"}'
+
+      substituteInPlace src/hhd/device/oxp/serial.py \
+        --replace-fail "udevadm" "${udevadm}"
+
+      substituteInPlace src/hhd/plugins/overlay/systemd.py \
+        --replace-fail "dbus-monitor" "${lib.getExe' dbus "dbus-monitor"}" \
+        --replace-fail "systemd-inhibit" "${lib.getExe' systemd "systemd-inhibit"}"
+
+      substituteInPlace src/hhd/plugins/overlay/x11.py \
+        --replace-fail "lsof" "${lib.getExe lsof}"
+
+      substituteInPlace src/hhd/plugins/plugin.py \
+        --replace-fail "id" "${lib.getExe' toybox "id"}"
```

There's [quite a bit](https://github.com/search?q=repo%3Ahhd-dev%2Fhhd+path%3Aplugin.py+id&type=code) of variables that have `id` on their name so this breaks handheld-daemon.
```suggestion
        --replace-fail '"id"' '"${lib.getExe' toybox "id"}"'
```
This change would make it only change the path of `id`, which is what we want

---

Good catch, done.